Feature - add message context filter to the message handler registration (#91)
arcus-azure/arcus.messaging: docs/preview/features/message-pumps/customization.md

}
```

> ## Control message context filtering

Reviewer: What do you think of one of these?
```suggestion
## Filter messages based on message context
```

```suggestion
## Process subset of messages with message context filtering
```

> When registering a new `IMessageHandler<>`, one can choose to add an additional filter on the message context used in that handler.

Reviewer: What about this?
```suggestion
When registering a new message handler, one can opt-in to add a filter on the message context which filters out messages that are not needed to be processed.

This can be useful when you are sending different message types on the same queue. Another use-case is being able to handle different versions of the same message type which have different contracts because you are migrating your application.
```

Following example shows how a message handler should only process a certain message when a property's in the context is present.

We'll use a simple message handler implementation:

```csharp
public class OrderMessageHandler : IMessageHandler<Order>
{

>     public async Task ProcessMessageAsync(Order order, MessageContext context, ...)

Reviewer: It's been a while so here is a stupid question - They were able to use different MessageContext type as well, right? Like the Service bus one?

Author: Not a stupid question 😉 , yest that's true, one can register message handlers with custom and built-in message context, altough I didn't implement or test this just yet in this PR. I thought it was a good start to first begin with the 'baseline' for this message context filtering.

Also why this PR is not closing the related issue 🤓 .

Reviewer: Ok, thanks!

Author: But as explained below, I added some extensions for this too bc I needed it for the integration tests.

    {
        // Do some processing...
	}
}
```

We would like that this handler only processed the message when the context contains `MessageType` equals `Order`.

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.WithMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
}
```

> > Note that the order in which the message handlers are registered is importent in the message processing.

Reviewer: ```suggestion
> Note that the order in which the message handlers are registered is important in the message processing.
```
